### PR TITLE
docs: amend wrong links to async pipe

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

Some links in the aio guides containing the code `async` are wrongly redirected to the async pipe: https://angular.io/api/common/AsyncPipe

## What is the new behavior?

The links are either disabled using the `no-auto-link` class or redirect to the correct guide

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/deprecations.md [hunk 1]
```diff
@@ -99,7 +99,7 @@ This section contains a complete list all of the currently-deprecated APIs, with
...
-| [`async`](api/core/testing/async)                                                  | [`waitForAsync`](api/core/testing/waitForAsync)                                    | v11                   | The `async` function from `@angular/core/testing` has been renamed to `waitForAsync` in order to avoid confusion with the native JavaScript `async` syntax. The existing function is deprecated and will be removed in a future version. |
+| [`async`](api/core/testing/async)                                                  | [`waitForAsync`](api/core/testing/waitForAsync)                                    | v11                   | The [`async`](api/core/testing/async) function from `@angular/core/testing` has been renamed to `waitForAsync` in order to avoid confusion with the native JavaScript <code class="no-auto-link">async</code> syntax. The existing function is deprecated and will be removed in a future version. |
```
Comment (dario-piotrowicz) [inline aio/content/guide/deprecations.md:102]: ### Before:
[Screenshot at 2021-09-24 22-26-21](https://angular.io/guide/deprecations#angularcore)

### After:
Screenshot at 2021-09-24 22-25-01

Comment (dario-piotrowicz) [reply aio/content/guide/deprecations.md:102]: As you can see now the link redirects to the right doc page, unfortunately the color of the text changed... is the light blue text color only applied for links automatically created?

File: aio/content/guide/migration-update-module-and-target-compiler-options.md [hunk 2]
```diff
@@ -24,7 +24,7 @@ This migration provides improvements to the long-term supportability of projects
 For the functionality that executes on Node.js, such as Universal and Protractor, the new settings provide performance and troubleshooting benefits as well.
 The minimum Node.js version for the Angular CLI (v10.13) supports features in ES2018 and earlier.
 By targeting later ES versions, the compiler transforms less code and can use newer features directly.
-Since zone.js does not support native `async` and `await`, the universal builds still target ES2016.
+Since zone.js does not support native <code class="no-auto-link">async</code> and `await`, the universal builds still target ES2016.
```
Comment (dario-piotrowicz) [inline aio/content/guide/migration-update-module-and-target-compiler-options.md:27]: ### Before:
[Screenshot at 2021-09-24 22-35-22](https://angular.io/guide/migration-update-module-and-target-compiler-options#why-is-this-migration-necessary)

### After:
Screenshot at 2021-09-24 22-35-30

**Review by gkalpak — COMMENTED** [review]
Thx, @dario-piotrowicz!

The current changes look good to me, but while reviewing the PR it occurred to me that `async` is a common/generic word that can easily be included in backticks and can refer to several different things (the `async` pipe, the deprecated `async()` testing helper, the `async/await` JS feature or even found in code-snippets with `async` functions).

Doing a quick repo-wide regex search for `` `async\b `` revealed several more occurrences throughout the docs.

Therefore, I believe `async` is a good candidate for our [ignoreGenericWords][1] filter used with the [autoLinkCode][2] post-processor.

WDYT, @dario-piotrowicz?

[1]: https://github.com/angular/angular/blob/0220ec843870ac016496cf6629149b6285c477b7/aio/tools/transforms/angular-base-package/services/auto-link-filters/ignoreGenericWords.js#L8
[2]: https://github.com/angular/angular/blob/0220ec843870ac016496cf6629149b6285c477b7/aio/tools/transforms/angular-base-package/post-processors/auto-link-code.js

**Comment by dario-piotrowicz** [comment]
> Thx, @dario-piotrowicz!
> 
> The current changes look good to me, but while reviewing the PR it occurred to me that `async` is a common/generic word that can easily be included in backticks and can refer to several different things (the `async` pipe, the deprecated `async()` testing helper, the `async/await` JS feature or even found in code-snippets with `async` functions).
> 
> Doing a quick repo-wide regex search for `` `async\b `` revealed several more occurrences through out the docs.
> 
> Therefore, I believe `async` is a good candidate for our [ignoreGenericWords](https://github.com/angular/angular/blob/0220ec843870ac016496cf6629149b6285c477b7/aio/tools/transforms/angular-base-package/services/auto-link-filters/ignoreGenericWords.js#L8) filter used with the [autoLinkCode](https://github.com/angular/angular/blob/0220ec843870ac016496cf6629149b6285c477b7/aio/tools/transforms/angular-base-package/post-processors/auto-link-code.js) post-processor.
> 
> WDYT, @dario-piotrowicz?

Sorry I made a search for `` `async` `` instead of `` `async\b` `` or even `` `.*async.*` `` so I might have missed some :sweat: 

I just run a search with `` `.*async.*` `` but I could not actually find more wrong occurrences of the linking

For example I've found these:
Screenshot at 2021-09-25 12-08-26
Screenshot at 2021-09-25 12-07-09
Which are perfectly fine

I do agree that `async` is a generic term, but 
- I sort of think that in Angular development it is more linked to the async pipe rather than the native js async
- the `async` testing function is deprecated so, it shouldn't really come up that often 
- and based on this: https://github.com/angular/angular/blob/99f2ffc740279509b89bb96093e2515ce0b46ae1/aio/tools/transforms/angular-base-package/post-processors/auto-link-code.js#L100 which I assume is what does the splitting for the autolinking, other instances of functions or code in general containing "async" shouldn't often be considered for the linking

I also searched for `\sasync\s` in aio/content/examples and could not find anything that should have wrong links, mostly the keyword is used in e2e tests and when it's not it doesn't seem to be auto-linked:
Screenshot at 2021-09-25 12-38-05

So from what I said above, and based on the fact that usually I think we do want to get the auto-linking I think I would not put it in the `ignoreGenericWords` filter

Please let me know what you think and if I missed something (I would not be surprised if I did :sweat_smile:)

(either way if you think that it should go in the filter, I'll get right on it :slightly_smiling_face:, but I have a feeling that I might end up adding more manual links the async pipe than the current fixes for breaking unwanted links)

File: aio/content/guide/deprecations.md (hunk 1 above)
Comment (dario-piotrowicz) [reply aio/content/guide/deprecations.md:102]: Might have found a solution here: https://github.com/angular/angular/pull/43586 :slightly_smiling_face:

**Review by gkalpak — APPROVED** [review]
You are right, the rest of the occurrences are indeed referring to the `async` pipe (so do not need to be changed) 😇

There would be several incorrectly linked occurrences in code-snippets (extracted from example apps), but we are saved by the [filterPipes][1] filter (which I had forgotten about), which only considers pipes when they are the first word in the code-snippet or preceded by a `|` :tada:

So, you are right. Let's keep `async` out of the `ignoreGenericWords` filter 👍 

##
Reviewed-for: global-docs-approvers

[1]: https://github.com/angular/angular/blob/4555d8a31625564d457b93113ce8c71a24bdd472/aio/tools/transforms/angular-base-package/services/auto-link-filters/filterPipes.js

**Comment by dario-piotrowicz** [comment]
> You are right, the rest of the occurrences are indeed referring to the `async` pipe (so do not need to be changed) innocent
> 
> There would be several incorrectly linked occurrences in code-snippets (extracted from example apps), but we are saved by the [filterPipes](https://github.com/angular/angular/blob/4555d8a31625564d457b93113ce8c71a24bdd472/aio/tools/transforms/angular-base-package/services/auto-link-filters/filterPipes.js) filter (which I had forgotten about), which only considers pipes when they are the first word in the code-snippet or preceded by a `|` tada
> 
> So, you are right. Let's keep `async` out of the `ignoreGenericWords` filter +1
> 
> Reviewed-for: global-docs-approvers

I had no idea about the `filterPipes` filter, it's quite neat :smiley: 

Thanks! :smile:

**Pull request closed without merging**